### ISSUE TITLE
Jekyll doesn't live reload

### ISSUE THREAD

**Issue by NealHumphrey** [codefordc/project-management#30]
While running the `docker-compose up` from the root of housing-insights dev branch: github.com/codefordc/housing-insights

Have to restart docker compose for changes to Jekyll site to be displayed. 

I used the instructions to build a custom Jekyll image (using edits to instructions noted in issue #23)
 
This is the relevant part of the docker compose file:
```
  jekyll:
        image: codefordc2/housing-insights-jekyll
        command:
            - jekyll serve --watch --incremental
        volumes:
            - ./docs:/srv/jekyll
        ports:
            - "4000:4000"
```

When I run this command directly, instead of docker-compose, I also do not get live reloading:
`docker run --volume $(pwd)/docs:/srv/jekyll -p 192.168.99.100:4000:4000 codefordc2/housing-insights-jekyll`

I'm guessing the problem is probably in the docker image? Is there something in the [Jekyll Dockerfile](https://github.com/codefordc/project-management/blob/master/jekyll/Dockerfile) that needs to be changed to allow the image to use live reloading?

**Comment by mkalish** [comment]
@NealHumphrey 
I think the confusion was my misuse of the word live reload.  After changing a file for Jekyll, you'll need to refresh the page in the browser, but you should not need to restart docker.  

Its working for me locally on mac, but this might be another docker-machine issue where the file changes are not getting picked up so let me know.

**Comment by NealHumphrey** [comment]
No confusion on the wording - the problem is that after refreshing the page changes aren't shown. I also have Jekyll installed directly on my machine and live reloading works when I just use Jekyll serve.

This has only been on my machine, but I don't know if anyone else has tested this part. If it is working for you with the housing insights repo and image then there is probably an OS issue.

**Comment by NealHumphrey** [comment]
@mkalish when I run `jekyll serve` w/ my local jekyll installation, any time I save a file it prints the 'regenerating... ' message to the command line. With Docker this doesn't happen. Since it works on your machine, maybe it's an issue with binding the local filesystem to the Docker filesystem?

**Comment by mkalish** [comment]
Yea it seems like an issue with your virtual machine not picking up file changes. I'll try to figure out a solution, but anyone not running docker machine should not have an issue

**Comment by NealHumphrey** [comment]
Closing this issue - partial fix is implemented:
- --force-polling added to default image 
- I have added documentation on how to diagnose a slow rebuild, which seems to be the remaining issue, so at least users know when this problem is occuring. 
- Recommended native install when the problem occurs. 

I did further investigation to try to solve the root cause of this, which seems to be slow communication between the mounted volume. It does not seem to be something there's an easy fix  to yet - all the solutions I found required additional packages or software installs. Hopefully Docker will fix the issue on their end eventually. 

Possible solutions for ambitious developers:
- Mac: https://github.com/brikis98/docker-osx-dev
- All, but w/ compose.yml changes: https://github.com/EugenMayer/docker-sync
- another: https://github.com/cweagans/docker-bg-sync
